### PR TITLE
Get chat working after account reset

### PR DESCRIPTION
**Pull request by patrickxb**
This PR makes chat work after account reset in CLI, json api.

Some notes:

1. I have not tried this with the GUI.  The design specifies a different display than CLI where the messages before the account reset show up in the same conversation.  Will talk to those involved and figure out the state of all that.
1. the user who performed the account reset will not be able to read the old conversation due to kbfs rekeying policy
1. all of this testing was done manually with a hacked up local mdserver.  When KBFS-1821 completed, automated tests should be written as this is some hairy stuff...

### CODE REVIEW

File: go/chat/utils.go [hunk 1]
```diff
@@ -254,3 +254,20 @@ func IsVisibleChatMessageType(messageType chat1.MessageType) bool {
 	}
 	return false
 }
+
+func AppendTLFResetSuffix(msgs []chat1.MessageBoxed, finalizeInfo *chat1.ConversationFinalizeInfo) []chat1.MessageBoxed {
+	if finalizeInfo == nil {
+		return msgs
+	}
+
+	if len(finalizeInfo.ResetFull) == 0 {
```
Comment (mmaxim) [inline go/chat/utils.go]: Is this to handle cases where Gregor was not recording this info?

File: go/service/tlf.go [hunk 2]
```diff
@@ -65,6 +66,8 @@ func (h *tlfHandler) CryptKeys(ctx context.Context, arg keybase1.TLFQuery) (keyb
 		return keybase1.GetTLFCryptKeysRes{}, err
 	}
 
+	debug.PrintStack()
```
Comment (mmaxim) [inline go/service/tlf.go]: Mistake?

File: go/protocol/chat1/extras.go [hunk 3]
```diff
@@ -224,3 +225,47 @@ func (q *GetInboxQuery) Visibility() TLFVisibility {
...
+	if len(finalizeInfo.ResetFull) == 0 {
+		return name
+	}
+	if strings.Contains(name, " account reset ") {
+		return name
+	}
+	return name + " " + finalizeInfo.ResetFull
+}
+
+// BeforeSummary returns a summary of the finalize without "files" in it.
```
Comment (mmaxim) [inline go/protocol/chat1/extras.go]: Yeah this is a nice consideration.

File: go/chat/convsource.go [hunk 4]
```diff
@@ -393,15 +403,18 @@ func (s *HybridConversationSource) GetMessages(ctx context.Context, convID chat1
 	// Form final result
 	var res []chat1.MessageUnboxed
 	for index, msg := range msgs {
+		var m chat1.MessageUnboxed
```
Comment (mmaxim) [inline go/chat/convsource.go]: Why did you make this change?

File: go/chat/utils.go [hunk 5]
```diff
@@ -254,3 +254,20 @@ func IsVisibleChatMessageType(messageType chat1.MessageType) bool {
...
+		return msgs
+	}
+
+	mod := make([]chat1.MessageBoxed, len(msgs))
+	for i, m := range msgs {
+		m.ClientHeader.TlfName = m.ClientHeader.TLFNameExpanded(finalizeInfo)
```
Comment (mmaxim) [inline go/chat/utils.go]: Maybe have an `ExpandTLF` function on `MessageClientHeader`?

**Comment by mmaxim** [comment]
Overall, looks good, much shorter than I thought it would be, which is good.

**Comment by mmaxim** [comment]
Also, all that `supersedes` and `supersededBy` business on `Conversation` is supposed to drive the GUI display.

File: go/chat/utils.go (hunk 1 above)
Comment (patrickxb) [inline go/chat/utils.go]: yes

File: go/service/tlf.go (hunk 2 above)
Comment (patrickxb) [inline go/service/tlf.go]: yup, thanks

File: go/chat/convsource.go (hunk 4 above)
Comment (patrickxb) [inline go/chat/convsource.go]: sorry, that is an artifact of a previous change.  I'll clean it up to the original.

File: go/chat/utils.go (hunk 5 above)
Comment (patrickxb) [inline go/chat/utils.go]: there is, that's what is being called.  Am I misunderstanding?

Comment (mmaxim) [inline go/chat/utils.go]: Hmm yeah I guess this is as good as you can do, I was thinking something like:

`m.ClientHeader.ExpandTLF()`

**Comment by patrickxb** [comment]
Yeah, there was a previous PR that had some of the work to fix this up which makes this somewhat smaller.

**Comment by mmaxim** [comment]
I guess it is pretty hard to write tests for this, but is there any room for some unit tests? Especially with respect to tagging the suffix onto the TLF names. We can always make that an addition ticket as well if it is too annoying.

File: go/chat/utils.go (hunk 5 above)
Comment (patrickxb) [inline go/chat/utils.go]: That works on ConversationInfoLocal, but not ClientHeader...

Comment (mmaxim) [inline go/chat/utils.go]: Sounds good!

**Comment by patrickxb** [comment]
We could write some tests in gregor easily that test ConversationFinalizeInfo and the stuff I added to chat1/extras.go.

**Comment by mmaxim** [comment]
There should be tests in Gregor for that already, but if you think there is something missing we could do that too. I guess the hardest part is reading the finalized TLFs which needs the full setup to test though.

**Pull request merged**